### PR TITLE
Fix some sorting methods to operate correctly on slices

### PR DESCRIPTION
**Pull request by bal-a**
Pull request for the issue: https://github.com/scala/bug/issues/12568

The issue relates to `from` and `to` arguments not being propagated when sorting some type of collections.

Since this is my first PR I apologise for all the obvious things that I've missed, please point them out and I'll fix it promptly :)

### CODE REVIEW

**Comment by bal-a** [comment]
I signed the CLA

**Comment by bal-a** [comment]
The test that I added is not testing much - I'll add something better tomorrow

File: test/junit/scala/util/SortingTest.scala [hunk 1]
```diff
@@ -34,10 +36,17 @@ class SortingTest {
...
+    val xxs = Array.fill(size)(rng.nextInt(variety))
+    val sortSliceArray = xxs.clone(); Sorting.stableSort(sortSliceArray, xxs.length / 2, xxs.length)
+    val sliceMin = xxs.slice(xxs.length / 2, xxs.length).minOption.getOrElse(Int.MinValue)
+    val sliceAboveMinCountsEqual = xxs.slice(xxs.length / 2, xxs.length).count(_ >= sliceMin) == sortSliceArray.slice(xxs.length / 2, xxs.length).count(_ >= sliceMin)
     
     assertTrue("Quicksort should be in order", isSorted(qxs))
     assertTrue("Quicksort should be in reverse order", isAntisorted(pxs))
     assertTrue("Stable sort should be sorted and stable", isStable(sxs))
+    assertTrue("Stable sort on a slice should produce a sorted slice", isSorted(sortSliceArray, xxs.length / 2 , xxs.length))
+    assertTrue("Stable sort on a slice should produce a slice with the same count of elems larger than the slice min", sliceAboveMinCountsEqual)
```
Comment (bal-a) [inline test/junit/scala/util/SortingTest.scala]: Tried not to deviate too much from the existing formatting in the test suite - this makes sure that the slice is sorted, as well as the number of elems over the min in the slice is the same, which ensures that we're not sorting the whole thing

File: test/junit/scala/util/SortingTest.scala [hunk 2]
```diff
@@ -34,10 +36,17 @@ class SortingTest {
     val sxs = { val temp = xs.clone; Sorting.stableSort(temp); temp }
     val rxs = { val temp = xs.clone; Sorting.stableSort(temp)(backwardsN); temp }
     val sys = Sorting.stableSort(ys.clone.toIndexedSeq, (i: Int) => xs(i))
+
+    val xxs = Array.fill(size)(rng.nextInt(variety))
+    val sortSliceArray = xxs.clone(); Sorting.stableSort(sortSliceArray, xxs.length / 2, xxs.length)
+    val sliceMin = xxs.slice(xxs.length / 2, xxs.length).minOption.getOrElse(Int.MinValue)
```
Comment (bal-a) [inline test/junit/scala/util/SortingTest.scala:42]: minOption is needed to make sure the test suite works on an empty input

File: test/junit/scala/util/SortingTest.scala [hunk 3]
```diff
@@ -34,10 +36,17 @@ class SortingTest {
...
+    val sliceAboveMinCountsEqual = xxs.slice(xxs.length / 2, xxs.length).count(_ >= sliceMin) == sortSliceArray.slice(xxs.length / 2, xxs.length).count(_ >= sliceMin)
```
Comment (NthPortal) [inline test/junit/scala/util/SortingTest.scala:43]: could we also check that the first half of the arrays are identical?

**Review by NthPortal — APPROVED** [review]

File: test/junit/scala/util/SortingTest.scala (hunk 3 above)
Comment (lrytz) [reply test/junit/scala/util/SortingTest.scala:43]: @bal-a send a ping once that's pushed, then I'll merge the PR. You can squash into the second commit. Thanks!

Comment (bal-a) [reply test/junit/scala/util/SortingTest.scala:43]: Thank you both for the feedback - added it and squashed it with the second commit, let me know if something looks off :)

@lrytz ping

**Comment by SethTisue** [comment]
@bal-a Thank you!

**Pull request merged**